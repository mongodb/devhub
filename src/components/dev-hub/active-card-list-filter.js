import React from 'react';
import CardList from './card-list';
import { LearnPageTabs } from '../../utils/learn-page-tabs';

const ActiveCardListFilter = ({ activeContentTab, textFilterResults }) => {
    switch (activeContentTab) {
        case LearnPageTabs.articles:
            return <CardList articles={textFilterResults} shouldSort />;
        case LearnPageTabs.videos:
            return <CardList videos={textFilterResults} shouldSort />;
        case LearnPageTabs.podcasts:
            return <CardList podcasts={textFilterResults} shouldSort />;
        default:
            return <CardList all={textFilterResults} shouldSort />;
    }
};

export default ActiveCardListFilter;
