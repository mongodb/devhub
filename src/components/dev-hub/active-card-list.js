import React from 'react';
import CardList from './card-list';
import { LearnPageTabs } from '../../utils/learn-page-tabs';

const ActiveCardList = ({
    activeContentTab,
    articles,
    videos,
    podcasts,
    hasNoFilter,
}) => {
    switch (activeContentTab) {
        case LearnPageTabs.articles:
            return <CardList articles={articles} shouldSort />;
        case LearnPageTabs.videos:
            return <CardList videos={videos} shouldSort />;
        case LearnPageTabs.podcasts:
            return <CardList podcasts={podcasts} shouldSort />;
        default:
            return (
                <CardList
                    shouldSort
                    articles={articles}
                    videos={hasNoFilter ? videos : []}
                    podcasts={hasNoFilter ? podcasts : []}
                />
            );
    }
};

export default ActiveCardList;
