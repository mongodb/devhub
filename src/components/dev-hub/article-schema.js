import React from 'react';
import { Helmet } from 'react-helmet';

const getAuthorsList = authors =>
    authors?.map(({ name }) => ({
        '@type': 'Person',
        name,
    }));

const ArticleSchema = ({
    authors,
    articleUrl,
    description,
    imageUrl,
    modifiedDate,
    publishedDate,
    tags,
    title,
    language = 'English',
}) => (
    <Helmet>
        <script id="structured-data" type="application/ld+json">
            {JSON.stringify({
                '@context': 'https://schema.org',
                '@type': 'BlogPosting',
                url: articleUrl,
                headline: title,
                description,
                datePublished: publishedDate,
                dateModified: modifiedDate || publishedDate,
                mainEntityOfPage: {
                    '@type': 'WebPage',
                    url: articleUrl,
                },
                image: {
                    '@type': 'imageObject',
                    url: imageUrl,
                },
                publisher: {
                    '@type': 'Organization',
                    name: 'MongoDB',
                    logo: {
                        '@type': 'imageObject',
                        url: 'https://webassets.mongodb.com/_com_assets/cms/mongodb_logo1-76twgcu2dm.png',
                    },
                },
                author: getAuthorsList(authors),
                tags,
                inLanguage: language,
            })}
        </script>
    </Helmet>
);

export default ArticleSchema;
