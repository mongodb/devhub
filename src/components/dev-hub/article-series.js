import React from 'react';
import dlv from 'dlv';
import Series from './series';

const ArticleSeries = ({ allSeriesForArticle, slugTitleMapping, title }) => {
    // Handle if this article is not in a series or no series are defined
    if (!allSeriesForArticle) return null;
    const getMappedSeries = seriesSlugs => {
        if (!seriesSlugs || !seriesSlugs.length) return null;

        let hasSeenActiveSlug = false;
        const getSeriesArticlePosition = articleTitle => {
            // Find if an article is the current one in the series, or is upcoming/past
            if (hasSeenActiveSlug) {
                return 'upcoming';
            } else if (articleTitle === title) {
                hasSeenActiveSlug = true;
                return 'active';
            }
            return 'past';
        };

        const mappedSeries = seriesSlugs.map(slug => {
            const title = dlv(slugTitleMapping, [slug, 0, 'value'], slug);
            const position = getSeriesArticlePosition(title);
            return {
                position,
                slug,
                title,
            };
        });
        return mappedSeries;
    };

    return Object.keys(allSeriesForArticle).map(series => {
        const seriesItems = getMappedSeries(allSeriesForArticle[series]);
        return <Series name={series}>{seriesItems}</Series>;
    });
};

export default ArticleSeries;
