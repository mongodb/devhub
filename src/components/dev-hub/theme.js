import { css } from '@emotion/react';

/**
 * @type {Object.<string, string>}
 * @property {string} fontsize returns px value
 */
const fontSize = {
    micro: '10px',
    xsmall: '12px',
    tiny: '14px',
    small: '16px',
    default: '18px',
    medium: '20px',
    large: '24px',
    xlarge: '30px',
    xxlarge: '36px',
    jumbo: '42px',
};
/**
 * @type {Object.<string, string>}
 * @property {string} lineHeight
 */
const lineHeight = {
    micro: '16px',
    xsmall: '18px',
    tiny: '20px',
    small: '24px',
    default: '28px',
    medium: '32px',
    large: '36px',
    xlarge: '42px',
    xxlarge: '48px',
    jumbo: '54px',
};
/**
 * @type {Object}
 * @property {string} size returns px value
 */
const size = {
    tiny: '5px',
    xsmall: '8px',
    small: '10px',
    default: '16px',
    medium: '20px',
    mediumLarge: '24px',
    articleContent: '30px',
    large: '32px',
    xlarge: '64px',
    xxlarge: '120px',
    maxContentWidth: '780px',
    maxWidth: '1440px',
    /** @type {function(string): number} */
    stripUnit(unit) {
        return parseInt(unit, 10);
    },
};

const colorMap = {
    darkGreen: '#13AA52',
    devBlack: '#0C1C27',
    devWhite: '#f9fbfa',
    greyDarkOne: '#5D6C74',
    greyDarkTwo: '#3D4F58',
    greyDarkThree: '#21313C',
    greyLightOne: '#E7EEEC',
    greyLightTwo: '#B8C4C2',
    greyLightThree: '#9FA1A2',
    lightGreen: '#0AD05B',
    orange: '#F97216',
    magenta: '#D34F94',
    pageBackground: '#061621',
    salmon: '#E55F55',
    sherbet: '#F7A76F',
    teal: '#2F9FC5',
    violet: '#6E60F9',
    purple: '#4F4FBF',
    yellow: '#FFDD49',
};

const gradientMap = {
    green: `linear-gradient(
        315deg,
        ${colorMap.lightGreen} 0%,
        ${colorMap.lightGreen} 40%,
        ${colorMap.darkGreen} 100%
    )`,
    greenTeal: `linear-gradient(
        270deg,
        ${colorMap.lightGreen} 0%,
        ${colorMap.teal} 100%
    )`,
    greenTealOffset: `linear-gradient(
        315deg,
        ${colorMap.teal} 0%,
        ${colorMap.lightGreen} 100%
    )`,
    magentaSalmonSherbet: `linear-gradient(
        270deg,
        ${colorMap.sherbet} 0%,
        ${colorMap.salmon} 49.99%,
        ${colorMap.magenta} 100%
    )`,
    magentaSalmonYellow: `linear-gradient(
        270deg,
        ${colorMap.yellow} 0%,
        ${colorMap.salmon} 49.99%,
        ${colorMap.magenta} 100%
    )`,
    tealViolet: `linear-gradient(
        270deg,
        ${colorMap.teal} 0%,
        ${colorMap.violet} 100%
    )`,
    tealVioletReverse: `linear-gradient(
        270deg,
        ${colorMap.violet} 0%,
        ${colorMap.teal} 100%
    )`,
    tealVioletPurple: `linear-gradient(
        320deg, 
        ${colorMap.teal} 0%, 
        ${colorMap.violet} 33.14%, 
        ${colorMap.purple} 100%
    )`,
    violetMagenta: `linear-gradient(
        270deg,
        ${colorMap.violet} 0%,
        ${colorMap.magenta} 100%
    )`,
    violentMagentaOrange: `linear-gradient(
        270deg,
        ${colorMap.violet} 0%,
        ${colorMap.magenta} 49.99%,
        ${colorMap.orange} 100%
    )`,
    violetMagentaReverse: `linear-gradient(
        270deg,
        ${colorMap.magenta} 0%,
        ${colorMap.violet} 100%
    )`,
};

const borderGradients = {
    greenTeal: `
        border: 2px solid;
        border-image: ${gradientMap.greenTeal};
        border-image-slice: 1;
    `,
    tealViolet: `
        border: 2px solid;
        border-image: ${gradientMap.tealViolet};
        border-image-slice: 1;
    `,
    tealVioletReverse: `
        border: 2px solid;
        border-image: ${gradientMap.tealVioletReverse};
        border-image-slice: 1;
    `,
    violetMagenta: `
        border: 2px solid;
        border-image: ${gradientMap.violetMagenta};
        border-image-slice: 1;
    `,
    violetMagentaReverse: `
        border: 2px solid;
        border-image: ${gradientMap.violetMagentaReverse};
        border-image-slice: 1;
    `,
};

/**
 * store common responsive sizes
 * @type {Object.<string, string>}
 */
const screenSize = {
    upToSmall: 'only screen and (max-width: 420px)',
    smallAndUp: 'not all and (max-width: 420px)',
    upToMedium: 'only screen and (max-width: 767px)',
    mediumAndUp: 'not all and (max-width: 767px)',
    upToLarge: 'only screen and (max-width: 1023px)',
    largeAndUp: 'not all and (max-width: 1023px)',
    upToSmallDesktop: 'only screen and (max-width: 1200px)',
    smallDesktopAndUp: 'not all and (max-width: 1200px)',
    upToXlarge: `only screen and (max-width: ${size.maxWidth})`,
    xlargeAndUp: `not all and (max-width: ${size.maxWidth})`,
};

const gridLayout = {
    desktop: {
        columnGap: '24px',
        numCols: 12,
        sideMargin: size.xxlarge,
    },
    mobile: {
        columnGap: size.default,
        numCols: 4,
        sideMargin: size.default,
    },
    tablet: {
        columnGap: size.default,
        numCols: 8,
        sideMargin: size.xlarge,
    },
};

const grid = css`
    display: grid;
    grid-template-columns: repeat(${gridLayout.desktop.numCols}, 1fr);
    column-gap: ${gridLayout.desktop.columnGap};
    margin: 0 ${gridLayout.desktop.sideMargin};
    @media ${screenSize.upToLarge} {
        grid-template-columns: repeat(${gridLayout.tablet.numCols}, 1fr);
        column-gap: ${gridLayout.tablet.columnGap};
        margin: 0 ${gridLayout.tablet.sideMargin};
    }
    @media ${screenSize.upToMedium} {
        grid-template-columns: repeat(${gridLayout.mobile.numCols}, 1fr);
        column-gap: ${gridLayout.mobile.columnGap};
        margin: 0 ${gridLayout.mobile.sideMargin};
    }
`;

/**
 * z-index values in ranges of 10. This should give enough leeway to incremement in components as needed.
 * @type {Object.<string, number>}
 * @property {string} position returns a z-index value
 */
const layer = {
    superBack: -1,
    back: 1,
    middle: 11,
    front: 21,
    superFront: 22,
};

/**
 * @type {Object.<string, string>}
 * @property {string} animationSpeed
 */
const animationSpeed = {
    slow: '650ms',
    medium: '300ms',
    fast: '150ms',
};

const FORM_ELEMENT_BORDER = '2px';

const HERO_CONTENT_WIDTH = '640px';

// Use colors with emotion theme
export const darkTheme = {
    borderGradients,
    colorMap,
    gradientMap,
};

export {
    animationSpeed,
    borderGradients,
    colorMap,
    fontSize,
    FORM_ELEMENT_BORDER,
    gradientMap,
    grid,
    HERO_CONTENT_WIDTH,
    layer,
    lineHeight,
    screenSize,
    size,
};
