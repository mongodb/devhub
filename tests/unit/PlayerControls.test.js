import React from 'react';
import { render } from 'enzyme';

import { ThemeProvider } from '@emotion/react';
import { darkTheme } from '~components/dev-hub/theme';
import PlayerControls from '~components/dev-hub/podcast-player/player-contols';

describe('PlayerControls', () => {
    it('renders correctly play', () => {
        const tree = render(
            <ThemeProvider theme={darkTheme}>
                <PlayerControls isPlaying={false} />
            </ThemeProvider>
        );
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly pause', () => {
        const tree = render(
            <ThemeProvider theme={darkTheme}>
                <PlayerControls isPlaying={true} />
            </ThemeProvider>
        );
        expect(tree).toMatchSnapshot();
    });
});
