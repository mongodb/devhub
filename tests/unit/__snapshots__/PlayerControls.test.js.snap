// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerControls renders correctly pause 1`] = `
<div
  class="css-rrotlg-ControlsContainer ecyqnyh2"
>
  <a
    aria-label="rewind"
    class="ecyqnyh1 e1lzeyoj2 css-badypa-StyledLink-linkStyling-AsLink-StyledButton-tertiaryStyles-StyledTimeControlButton-buttonStyles eygprlc0"
    role="button"
    tabindex="0"
  >
    <svg
      fill="none"
      height="33px"
      width="31px"
    >
      <path
        clip-rule="evenodd"
        d="M8.33 4.36a1.25 1.25 0 000 2.16l5.56 3.23c.83.49 1.88-.11 1.88-1.08V6c6.3.4 11.3 5.68 11.3 12.15 0 6.74-5.41 12.18-12.07 12.18A12.12 12.12 0 012.93 18.14a1 1 0 10-2 0c0 7.83 6.3 14.18 14.07 14.18 7.78 0 14.07-6.35 14.07-14.18 0-7.56-5.88-13.75-13.3-14.15V2.2c0-.97-1.05-1.57-1.88-1.08L8.33 4.36zm5.44-.85v3.86l-3.32-1.93 3.32-1.93z"
        fill="url(#paint0_linear)"
        fill-rule="evenodd"
      />
      <path
        d="M11.7 13.9c.54 0 1.01.1 1.41.3.4.18.7.43.92.75.22.32.33.67.33 1.05 0 .5-.16.92-.47 1.26-.3.35-.7.58-1.18.69a2.2 2.2 0 011.37.6c.36.34.54.82.54 1.44 0 .47-.12.9-.37 1.28-.25.39-.6.7-1.06.91a3.97 3.97 0 01-3.08.04c-.47-.2-.86-.49-1.18-.86l.83-.72c.23.27.5.47.8.6.32.14.65.21 1 .21.51 0 .92-.13 1.21-.4.3-.28.46-.65.46-1.11 0-.97-.54-1.45-1.6-1.45h-.72l.16-.99h.5c.42 0 .76-.1 1.04-.33s.42-.55.42-.97c0-.4-.14-.7-.4-.92-.26-.22-.6-.33-1.03-.33-.34 0-.64.06-.91.18-.28.12-.54.3-.8.56l-.72-.77a3.52 3.52 0 012.54-1.02zm7.56 0c.92 0 1.62.38 2.1 1.12.48.74.72 1.8.72 3.18a5.9 5.9 0 01-.72 3.2 2.34 2.34 0 01-2.1 1.1c-.92 0-1.62-.36-2.1-1.1a5.9 5.9 0 01-.72-3.2c0-1.38.24-2.44.72-3.18a2.34 2.34 0 012.1-1.12zm0 1.04c-.48 0-.84.26-1.09.78a6.13 6.13 0 00-.37 2.48c0 1.15.12 1.99.37 2.5.25.53.61.79 1.1.79.47 0 .84-.26 1.08-.78.25-.52.38-1.36.38-2.5 0-1.15-.13-1.98-.38-2.5-.24-.51-.6-.77-1.09-.77zm0 2.5c.23 0 .41.07.57.22.15.15.22.34.22.55a.77.77 0 01-.8.8.77.77 0 01-.55-.23.79.79 0 01-.22-.57.77.77 0 01.78-.78z"
        fill="url(#paint1_linear)"
      />
      <defs>
        <linearGradient
          gradientUnits="userSpaceOnUse"
          id="paint0_linear"
          x1="28.83"
          x2="-2.11"
          y1="32.22"
          y2="4.33"
        >
          <stop
            stop-color="#F7A76F"
          />
          <stop
            offset=".38"
            stop-color="#E55F55"
          />
          <stop
            offset=".54"
            stop-color="#E55F55"
          />
          <stop
            offset=".73"
            stop-color="#D34F94"
          />
          <stop
            offset="1"
            stop-color="#D34F94"
          />
        </linearGradient>
        <linearGradient
          gradientUnits="userSpaceOnUse"
          id="paint1_linear"
          x1="23.34"
          x2="9.5"
          y1="25.31"
          y2="10.4"
        >
          <stop
            stop-color="#F7A76F"
          />
          <stop
            offset=".38"
            stop-color="#E55F55"
          />
          <stop
            offset=".54"
            stop-color="#E55F55"
          />
          <stop
            offset=".73"
            stop-color="#D34F94"
          />
          <stop
            offset="1"
            stop-color="#D34F94"
          />
        </linearGradient>
      </defs>
    </svg>
  </a>
  <a
    aria-label="pause"
    class="ecyqnyh0 e1lzeyoj2 css-154jfdt-StyledLink-linkStyling-AsLink-StyledButton-tertiaryStyles-StyledMainButton-buttonStyles eygprlc0"
    role="button"
    tabindex="0"
  >
    <svg
      fill="none"
      height="80px"
      viewBox="0 0 80px 80px"
      width="80px"
    >
      <circle
        cx="40"
        cy="40"
        r="39"
        stroke="url(#pause-icon-linear-1)"
        stroke-width="2"
      />
      <path
        d="M47.33 25.44v29.12"
        stroke="url(#pause-icon-linear-2)"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
      />
      <path
        d="M34 25.44v29.12"
        stroke="url(#pause-icon-linear-3)"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
      />
      <defs>
        <linearGradient
          gradientUnits="userSpaceOnUse"
          id="pause-icon-linear-1"
          x1="78.34"
          x2="1"
          y1="78.74"
          y2="1"
        >
          <stop
            stop-color="#F7A76F"
          />
          <stop
            offset=".38"
            stop-color="#E55F55"
          />
          <stop
            offset=".54"
            stop-color="#E55F55"
          />
          <stop
            offset=".73"
            stop-color="#D34F94"
          />
          <stop
            offset="1"
            stop-color="#D34F94"
          />
        </linearGradient>
        <linearGradient
          gradientUnits="userSpaceOnUse"
          id="pause-icon-linear-2"
          x1="48.33"
          x2="39.18"
          y1="55.56"
          y2="26.72"
        >
          <stop
            stop-color="#F7A76F"
          />
          <stop
            offset=".38"
            stop-color="#E55F55"
          />
          <stop
            offset=".54"
            stop-color="#E55F55"
          />
          <stop
            offset=".73"
            stop-color="#D34F94"
          />
          <stop
            offset="1"
            stop-color="#D34F94"
          />
        </linearGradient>
        <linearGradient
          gradientUnits="userSpaceOnUse"
          id="pause-icon-linear-3"
          x1="35"
          x2="25.84"
          y1="55.56"
          y2="26.72"
        >
          <stop
            stop-color="#F7A76F"
          />
          <stop
            offset=".38"
            stop-color="#E55F55"
          />
          <stop
            offset=".54"
            stop-color="#E55F55"
          />
          <stop
            offset=".73"
            stop-color="#D34F94"
          />
          <stop
            offset="1"
            stop-color="#D34F94"
          />
        </linearGradient>
      </defs>
    </svg>
  </a>
  <a
    aria-label="fast-forward"
    class="ecyqnyh1 e1lzeyoj2 css-badypa-StyledLink-linkStyling-AsLink-StyledButton-tertiaryStyles-StyledTimeControlButton-buttonStyles eygprlc0"
    role="button"
    tabindex="0"
  >
    <svg
      fill="none"
      height="33px"
      width="31px"
    >
      <path
        clip-rule="evenodd"
        d="M21.67 4.36c.82.48.82 1.68 0 2.16L16.1 9.75a1.25 1.25 0 01-1.88-1.08V6c-6.3.4-11.3 5.68-11.3 12.15 0 6.74 5.41 12.18 12.07 12.18s12.07-5.44 12.07-12.18a1 1 0 112 0c0 7.83-6.3 14.18-14.07 14.18A14.12 14.12 0 01.93 18.14C.93 10.58 6.81 4.4 14.23 4V2.2c0-.97 1.05-1.57 1.88-1.08l5.56 3.23zm-5.44-.85v3.86l3.32-1.93-3.32-1.93z"
        fill="url(#paint0_linear)"
        fill-rule="evenodd"
      />
      <path
        d="M11.7 13.9c.54 0 1.01.1 1.41.3.4.18.7.43.92.75.22.32.33.67.33 1.05 0 .5-.16.92-.47 1.26-.3.35-.7.58-1.18.69a2.2 2.2 0 011.37.6c.36.34.54.82.54 1.44 0 .47-.12.9-.37 1.28-.25.39-.6.7-1.06.91a3.97 3.97 0 01-3.08.04c-.47-.2-.86-.49-1.18-.86l.83-.72c.23.27.5.47.8.6.32.14.65.21 1 .21.51 0 .92-.13 1.21-.4.3-.28.46-.65.46-1.11 0-.97-.54-1.45-1.6-1.45h-.72l.16-.99h.5c.42 0 .76-.1 1.04-.33s.42-.55.42-.97c0-.4-.14-.7-.4-.92-.26-.22-.6-.33-1.03-.33-.34 0-.64.06-.91.18-.28.12-.54.3-.8.56l-.72-.77a3.52 3.52 0 012.54-1.02zm7.56 0c.92 0 1.62.38 2.1 1.12.48.74.72 1.8.72 3.18a5.9 5.9 0 01-.72 3.2 2.34 2.34 0 01-2.1 1.1c-.92 0-1.62-.36-2.1-1.1a5.9 5.9 0 01-.72-3.2c0-1.38.24-2.44.72-3.18a2.34 2.34 0 012.1-1.12zm0 1.04c-.48 0-.84.26-1.09.78a6.13 6.13 0 00-.37 2.48c0 1.15.12 1.99.37 2.5.25.53.61.79 1.1.79.47 0 .84-.26 1.08-.78.25-.52.38-1.36.38-2.5 0-1.15-.13-1.98-.38-2.5-.24-.51-.6-.77-1.09-.77zm0 2.5c.23 0 .41.07.57.22.15.15.22.34.22.55a.77.77 0 01-.8.8.77.77 0 01-.55-.23.79.79 0 01-.22-.57.77.77 0 01.78-.78z"
        fill="url(#paint1_linear)"
      />
      <defs>
        <linearGradient
          gradientUnits="userSpaceOnUse"
          id="paint0_linear"
          x1="1.17"
          x2="32.11"
          y1="32.22"
          y2="4.33"
        >
          <stop
            stop-color="#F7A76F"
          />
          <stop
            offset=".38"
            stop-color="#E55F55"
          />
          <stop
            offset=".54"
            stop-color="#E55F55"
          />
          <stop
            offset=".73"
            stop-color="#D34F94"
          />
          <stop
            offset="1"
            stop-color="#D34F94"
          />
        </linearGradient>
        <linearGradient
          gradientUnits="userSpaceOnUse"
          id="paint1_linear"
          x1="23.34"
          x2="9.5"
          y1="25.31"
          y2="10.4"
        >
          <stop
            stop-color="#F7A76F"
          />
          <stop
            offset=".38"
            stop-color="#E55F55"
          />
          <stop
            offset=".54"
            stop-color="#E55F55"
          />
          <stop
            offset=".73"
            stop-color="#D34F94"
          />
          <stop
            offset="1"
            stop-color="#D34F94"
          />
        </linearGradient>
      </defs>
    </svg>
  </a>
</div>
`;

exports[`PlayerControls renders correctly play 1`] = `
<div
  class="css-rrotlg-ControlsContainer ecyqnyh2"
>
  <a
    aria-label="rewind"
    class="ecyqnyh1 e1lzeyoj2 css-badypa-StyledLink-linkStyling-AsLink-StyledButton-tertiaryStyles-StyledTimeControlButton-buttonStyles eygprlc0"
    role="button"
    tabindex="0"
  >
    <svg
      fill="none"
      height="33px"
      width="31px"
    >
      <path
        clip-rule="evenodd"
        d="M8.33 4.36a1.25 1.25 0 000 2.16l5.56 3.23c.83.49 1.88-.11 1.88-1.08V6c6.3.4 11.3 5.68 11.3 12.15 0 6.74-5.41 12.18-12.07 12.18A12.12 12.12 0 012.93 18.14a1 1 0 10-2 0c0 7.83 6.3 14.18 14.07 14.18 7.78 0 14.07-6.35 14.07-14.18 0-7.56-5.88-13.75-13.3-14.15V2.2c0-.97-1.05-1.57-1.88-1.08L8.33 4.36zm5.44-.85v3.86l-3.32-1.93 3.32-1.93z"
        fill="url(#paint0_linear)"
        fill-rule="evenodd"
      />
      <path
        d="M11.7 13.9c.54 0 1.01.1 1.41.3.4.18.7.43.92.75.22.32.33.67.33 1.05 0 .5-.16.92-.47 1.26-.3.35-.7.58-1.18.69a2.2 2.2 0 011.37.6c.36.34.54.82.54 1.44 0 .47-.12.9-.37 1.28-.25.39-.6.7-1.06.91a3.97 3.97 0 01-3.08.04c-.47-.2-.86-.49-1.18-.86l.83-.72c.23.27.5.47.8.6.32.14.65.21 1 .21.51 0 .92-.13 1.21-.4.3-.28.46-.65.46-1.11 0-.97-.54-1.45-1.6-1.45h-.72l.16-.99h.5c.42 0 .76-.1 1.04-.33s.42-.55.42-.97c0-.4-.14-.7-.4-.92-.26-.22-.6-.33-1.03-.33-.34 0-.64.06-.91.18-.28.12-.54.3-.8.56l-.72-.77a3.52 3.52 0 012.54-1.02zm7.56 0c.92 0 1.62.38 2.1 1.12.48.74.72 1.8.72 3.18a5.9 5.9 0 01-.72 3.2 2.34 2.34 0 01-2.1 1.1c-.92 0-1.62-.36-2.1-1.1a5.9 5.9 0 01-.72-3.2c0-1.38.24-2.44.72-3.18a2.34 2.34 0 012.1-1.12zm0 1.04c-.48 0-.84.26-1.09.78a6.13 6.13 0 00-.37 2.48c0 1.15.12 1.99.37 2.5.25.53.61.79 1.1.79.47 0 .84-.26 1.08-.78.25-.52.38-1.36.38-2.5 0-1.15-.13-1.98-.38-2.5-.24-.51-.6-.77-1.09-.77zm0 2.5c.23 0 .41.07.57.22.15.15.22.34.22.55a.77.77 0 01-.8.8.77.77 0 01-.55-.23.79.79 0 01-.22-.57.77.77 0 01.78-.78z"
        fill="url(#paint1_linear)"
      />
      <defs>
        <linearGradient
          gradientUnits="userSpaceOnUse"
          id="paint0_linear"
          x1="28.83"
          x2="-2.11"
          y1="32.22"
          y2="4.33"
        >
          <stop
            stop-color="#F7A76F"
          />
          <stop
            offset=".38"
            stop-color="#E55F55"
          />
          <stop
            offset=".54"
            stop-color="#E55F55"
          />
          <stop
            offset=".73"
            stop-color="#D34F94"
          />
          <stop
            offset="1"
            stop-color="#D34F94"
          />
        </linearGradient>
        <linearGradient
          gradientUnits="userSpaceOnUse"
          id="paint1_linear"
          x1="23.34"
          x2="9.5"
          y1="25.31"
          y2="10.4"
        >
          <stop
            stop-color="#F7A76F"
          />
          <stop
            offset=".38"
            stop-color="#E55F55"
          />
          <stop
            offset=".54"
            stop-color="#E55F55"
          />
          <stop
            offset=".73"
            stop-color="#D34F94"
          />
          <stop
            offset="1"
            stop-color="#D34F94"
          />
        </linearGradient>
      </defs>
    </svg>
  </a>
  <a
    aria-label="play"
    class="ecyqnyh0 e1lzeyoj2 css-154jfdt-StyledLink-linkStyling-AsLink-StyledButton-tertiaryStyles-StyledMainButton-buttonStyles eygprlc0"
    role="button"
    tabindex="0"
  >
    <svg
      fill="none"
      height="80px"
      width="80px"
    >
      <circle
        cx="40"
        cy="40"
        r="39"
        stroke="url(#play-icon-linear-1)"
        stroke-width="2"
      />
      <path
        d="M56.7 39.16a1 1 0 010 1.68L32.89 56.32a1 1 0 01-1.55-.84V24.52a1 1 0 011.55-.84L56.7 39.16z"
        stroke="url(#play-icon-linear-2)"
        stroke-width="2"
      />
      <defs>
        <linearGradient
          gradientUnits="userSpaceOnUse"
          id="play-icon-linear-1"
          x1="78.34"
          x2="1"
          y1="78.74"
          y2="1"
        >
          <stop
            stop-color="#F7A76F"
          />
          <stop
            offset=".38"
            stop-color="#E55F55"
          />
          <stop
            offset=".54"
            stop-color="#E55F55"
          />
          <stop
            offset=".73"
            stop-color="#D34F94"
          />
          <stop
            offset="1"
            stop-color="#D34F94"
          />
        </linearGradient>
        <linearGradient
          gradientUnits="userSpaceOnUse"
          id="play-icon-linear-2"
          x1="49.11"
          x2="26.89"
          y1="80"
          y2="14.44"
        >
          <stop
            stop-color="#F7A76F"
          />
          <stop
            offset=".38"
            stop-color="#E55F55"
          />
          <stop
            offset=".54"
            stop-color="#E55F55"
          />
          <stop
            offset=".73"
            stop-color="#D34F94"
          />
          <stop
            offset="1"
            stop-color="#D34F94"
          />
        </linearGradient>
      </defs>
    </svg>
  </a>
  <a
    aria-label="fast-forward"
    class="ecyqnyh1 e1lzeyoj2 css-badypa-StyledLink-linkStyling-AsLink-StyledButton-tertiaryStyles-StyledTimeControlButton-buttonStyles eygprlc0"
    role="button"
    tabindex="0"
  >
    <svg
      fill="none"
      height="33px"
      width="31px"
    >
      <path
        clip-rule="evenodd"
        d="M21.67 4.36c.82.48.82 1.68 0 2.16L16.1 9.75a1.25 1.25 0 01-1.88-1.08V6c-6.3.4-11.3 5.68-11.3 12.15 0 6.74 5.41 12.18 12.07 12.18s12.07-5.44 12.07-12.18a1 1 0 112 0c0 7.83-6.3 14.18-14.07 14.18A14.12 14.12 0 01.93 18.14C.93 10.58 6.81 4.4 14.23 4V2.2c0-.97 1.05-1.57 1.88-1.08l5.56 3.23zm-5.44-.85v3.86l3.32-1.93-3.32-1.93z"
        fill="url(#paint0_linear)"
        fill-rule="evenodd"
      />
      <path
        d="M11.7 13.9c.54 0 1.01.1 1.41.3.4.18.7.43.92.75.22.32.33.67.33 1.05 0 .5-.16.92-.47 1.26-.3.35-.7.58-1.18.69a2.2 2.2 0 011.37.6c.36.34.54.82.54 1.44 0 .47-.12.9-.37 1.28-.25.39-.6.7-1.06.91a3.97 3.97 0 01-3.08.04c-.47-.2-.86-.49-1.18-.86l.83-.72c.23.27.5.47.8.6.32.14.65.21 1 .21.51 0 .92-.13 1.21-.4.3-.28.46-.65.46-1.11 0-.97-.54-1.45-1.6-1.45h-.72l.16-.99h.5c.42 0 .76-.1 1.04-.33s.42-.55.42-.97c0-.4-.14-.7-.4-.92-.26-.22-.6-.33-1.03-.33-.34 0-.64.06-.91.18-.28.12-.54.3-.8.56l-.72-.77a3.52 3.52 0 012.54-1.02zm7.56 0c.92 0 1.62.38 2.1 1.12.48.74.72 1.8.72 3.18a5.9 5.9 0 01-.72 3.2 2.34 2.34 0 01-2.1 1.1c-.92 0-1.62-.36-2.1-1.1a5.9 5.9 0 01-.72-3.2c0-1.38.24-2.44.72-3.18a2.34 2.34 0 012.1-1.12zm0 1.04c-.48 0-.84.26-1.09.78a6.13 6.13 0 00-.37 2.48c0 1.15.12 1.99.37 2.5.25.53.61.79 1.1.79.47 0 .84-.26 1.08-.78.25-.52.38-1.36.38-2.5 0-1.15-.13-1.98-.38-2.5-.24-.51-.6-.77-1.09-.77zm0 2.5c.23 0 .41.07.57.22.15.15.22.34.22.55a.77.77 0 01-.8.8.77.77 0 01-.55-.23.79.79 0 01-.22-.57.77.77 0 01.78-.78z"
        fill="url(#paint1_linear)"
      />
      <defs>
        <linearGradient
          gradientUnits="userSpaceOnUse"
          id="paint0_linear"
          x1="1.17"
          x2="32.11"
          y1="32.22"
          y2="4.33"
        >
          <stop
            stop-color="#F7A76F"
          />
          <stop
            offset=".38"
            stop-color="#E55F55"
          />
          <stop
            offset=".54"
            stop-color="#E55F55"
          />
          <stop
            offset=".73"
            stop-color="#D34F94"
          />
          <stop
            offset="1"
            stop-color="#D34F94"
          />
        </linearGradient>
        <linearGradient
          gradientUnits="userSpaceOnUse"
          id="paint1_linear"
          x1="23.34"
          x2="9.5"
          y1="25.31"
          y2="10.4"
        >
          <stop
            stop-color="#F7A76F"
          />
          <stop
            offset=".38"
            stop-color="#E55F55"
          />
          <stop
            offset=".54"
            stop-color="#E55F55"
          />
          <stop
            offset=".73"
            stop-color="#D34F94"
          />
          <stop
            offset="1"
            stop-color="#D34F94"
          />
        </linearGradient>
      </defs>
    </svg>
  </a>
</div>
`;
